sentry-types/src/protocol/envelope.rs

```diff
@@ -59,6 +59,64 @@ struct EnvelopeHeaders {
     trace: Option<DynamicSamplingContext>,
 }
 
+impl EnvelopeHeaders {
+    /// Creates empty Envelope headers.
+    pub fn new() -> EnvelopeHeaders {
+        Default::default()
+    }
+
+    /// Returns the Event ID.
+    pub fn event_id(&self) -> Option<&Uuid> {
+        self.event_id.as_ref()
+    }
+
+    /// Sets the Event ID.
+    pub fn set_event_id(&mut self, event_id: Option<Uuid>) {
```

Comment:
As discussed offline, you could consider using a builder pattern here:


```suggestion
    pub fn set_event_id(mut self, event_id: Option<Uuid>) -> Self {
```